Version 0.5 Release

rstoneback opened the pull request:
Implemented a new algorithm that produces first orthogonal geomagnetic basis vector system that works for non-Spherical planets as well as for multi-pole magnetic fields.

rstoneback committed:
ENH/BUG: Improved robustness and accuracy of step_until_intersect method.

rstoneback committed:
STY: Improved comments in step_until_intersect.

rstoneback committed:
ENH: Added tol keyword to step_until_intersect.

rstoneback committed:
BUG: Corrected number of sub-steps to ensure coverage of field_step_size in step_until_intersect.

rstoneback committed:
STY: Removed commented debug statements.

rstoneback committed:
BUG: Updated calls for the latest pysat standard.

rstoneback committed:
BUG: Updated pysat call to create_date_range.

rstoneback committed:
BUG: Updated pysat call to public release name season_date_range.

rstoneback committed:
BBUG: Atlas no longer in python3. Altered travis.yml.

jklenzing committed:
DOC: Add community docs

jklenzing committed:
DOC: update changelog

pulled out changes from other branch

rstoneback committed:
Merge pull request #22 from jklenzing/comm_docs

DOC: Add community docs

jklenzing committed:
ENH: update travis with no_sgp4 version

jklenzing committed:
DOC: update to new project badges

jklenzing committed:
DOC: spelling

jklenzing committed:
BUG: update pysat function getyrdoy

jklenzing committed:
BUG: add atlas to travis

jklenzing committed:
Update f2py setup

jklenzing committed:
BUG: update links to pysat organization

jklenzing committed:
BUG: extension name

jklenzing committed:
BUG: try python-numpy

jklenzing committed:
BUG: go back to conda in travis, keep no_sgp4

jklenzing committed:
BUG: update create_date_range

jklenzing committed:
BUG: remove atlas

jklenzing committed:
BUG: update create_date_range properly

jklenzing committed:
TST: try new travis environment

jklenzing committed:
STY: import from pysatMagVect, not .

jklenzing committed:
TST: clean travis

jklenzing committed:
BUG: typo in import

rstoneback committed:
Merge pull request #24 from pysat/travis_update

BUG/TST: Improved Compatibility with pysat and Travis CI

rstoneback committed:
Merge branch 'develop' into field_trace_fix

rstoneback committed:
BUG: Corrected IGRF call to use geocentric values

rstoneback committed:
BUG: Corrected date calculation

rstoneback committed:
BUG: Corrected date calculation. Restored geodetic.

rstoneback committed:
ENH: New localized calculation of unit vectors

rstoneback committed:
ENH: Matched step_size to unit_vector length scale

rstoneback committed:
STY: Enhanced compat and docstring for new method

rstoneback committed:
TES:Updated tests for new unit vector method.

rstoneback committed:
STY:Removed docstring text.

rstoneback committed:
TES: Added checks on input and output.

rstoneback committed:
ENH: Added fortran ECEF_to_geodetic method.

rstoneback committed:
TES: Added test for unit_vector algorithm sensitivity.

rstoneback committed:
BUG: Corrected variable use

rstoneback committed:
ENH: Added option for step sizes in apex_location_info

rstoneback committed:
BUG: Fortran func always returns array.

rstoneback committed:
BUG: Fixed setup.py fortran extensions

rstoneback committed:
BUG: Adapted to np.array output from Fortran

rstoneback committed:
ENH: Updated to latest method call standards.

rstoneback committed:
TES: Switched away from normalized difference, unit vector plots

rstoneback committed:
BUG: Updated for new method calls

rstoneback committed:
TES: Added test for apex location sensitivity

rstoneback committed:
ENH: Optimizing parameters

rstoneback committed:
TES: Added test for step size and local unit vectors. Improved long plots.

rstoneback committed:
TES: Improved plotting over longitude

rstoneback committed:
ENH: First order performance improvements via parameter selection

rstoneback committed:
BUG: Restored test plotting exception for Travis

rstoneback committed:
BUG: Improved f2py directives

rstoneback committed:
BUG: Improved unit vector calculation when bx,by near zero.

rstoneback committed:
TES: Improved test methods

rstoneback committed:
ENH: Use default unit_vectors in step_along. Added intial_factor to step_until.

rstoneback committed:
BUG: Enforced consistent test settings, step_along_mag

rstoneback committed:
BUG: Replaced int with ceil for getting number of steps

rstoneback committed:
ENH: Shifted filter to tol/10 for calc_unit_vectors

rstoneback committed:
ENH/TES: Added additional spatial plox apex location vs fine step size

rstoneback committed:
ENH: Tweaked default method parameters

rstoneback committed:
BUG/ENH: Filtering of height caused bug. Added check on integral output message.

rstoneback committed:
TES: Moved check to if recurse

rstoneback committed:
BUG: Tweaking unit vector method.

rstoneback committed:
ENH: Improved unit vector default based upon testing.

rstoneback committed:
BUG/TES: Fixed parentheses and titles.

rstoneback committed:
ENH: Feedback when things don't converge.

rstoneback committed:
TES: Update

rstoneback committed:
ENH: Improved seed vector. Added option for more output. Parameter improvements.

rstoneback committed:
ENH/TES: Added new plots to cover new output option for unit vectors.

rstoneback committed:
TES: Added multi-processor support.

rstoneback committed:
TES/BUG: Fixed difference parameters and errant , .

rstoneback committed:
ENH: Tweaked parameters based upon testing

rstoneback committed:
ENH: Tweaked parameters based upon testing

rstoneback committed:
ENH: Feedback on location with maximum uncertainty.

rstoneback committed:
Switched to derivative from difference. More robust termination.

rstoneback committed:
Corrected Titles.

rstoneback committed:
Corrected Titles.

rstoneback committed:
TES: Improved robustness.

rstoneback committed:
ENH: Updated tolerance of change in apex height to try and improve robustness.

rstoneback committed:
ENH: Updated tolerance of change in apex height to try and improve robustness.

rstoneback committed:
ENH: Updated tolerance of change in apex height to try and improve robustness.

rstoneback committed:
TES: Restored tests for scalig along mag field methods.

rstoneback committed:
BUG: Removed errant comma

rstoneback committed:
BUG: geocentric -> geodetic in unit vector method.

rstoneback committed:
ENH: New method for getting apex height difference at footpoints

rstoneback committed:
ENH: Incorporated new method in determining ion drift scalars.

rstoneback committed:
STY: cleanup of comments and docstrings.

rstoneback committed:
ENH: Implemented binary search in apex_location. Reduced default tolerance.

rstoneback committed:
ENH: Updated default parameters and restored use of step_size for unit_vect in  step_along.

rstoneback committed:
STY: cleanup

rstoneback committed:
TES: commented out closed_loop tests.

rstoneback committed:
BUG: Corrected number of sub test points.

rstoneback committed:
BUG: Removed print

rstoneback committed:
ENH: Faster and more accurate settings. Increased accuracy apex_locations.

rstoneback committed:
tweak

rstoneback committed:
TES: Condensed tests

rstoneback committed:
TES/BUG: Fixed bugs

rstoneback committed:
TES: Added plot for normalized difference apex altitude.

rstoneback committed:
TES/BUG: Fixed bugs

rstoneback committed:
TES/BUG: Fixed bugs

rstoneback committed:
TES/BUG: Fixed bugs

rstoneback committed:
TES/BUG: Fixed bugs

rstoneback committed:
TES/BUG: Fixed bugs

rstoneback committed:
TES/BUG: Fixed bugs

rstoneback committed:
TES/BUG: Fixed bugs

rstoneback committed:
TES/ENH: Added addition tests for ecef-geodetic impacts.

rstoneback committed:
TES/BUG: Fixed bugs

rstoneback committed:
STY: Restored test default.

rstoneback committed:
TES/ENH: Added new test for sensitivity to changes in fine_max_steps.

rstoneback committed:
TES/BUG: Fixed bugs

rstoneback committed:
ENH: ECEF position path through apex_location. Minimizes ecef, geodetic.

rstoneback committed:
TES/ENH: New test ecef, geodetic python v fortran. Typo correction.

rstoneback committed:
BUG: Typo correction.

rstoneback committed:
BUG: Typo correction.

rstoneback committed:
ENH: Reduced field line tracing tolerance.

rstoneback committed:
ENH: Reduced unit vector step size.

rstoneback committed:
ENH: Magnetic field vector now a method and more accurate. Added ECEF option to unit vector

rstoneback committed:
BUG: Fixed case for ecef input

rstoneback committed:
ENH: Removed geodetic use in step_along

rstoneback committed:
Improved accuracy and reduced function calls.

rstoneback committed:
TES: Improved test values. Vectorized calls.

rstoneback committed:
BUG: Fixed access order for data.

rstoneback committed:
ENH: Vectorized scalar_mapping... modified output from step_along.

rstoneback committed:
STY: Removed commented code.

rstoneback committed:
STY: Removed heritage scalars_for_mapping.... code

rstoneback committed:
TES/BUG: Fixed steps goal array.

rstoneback committed:
ENH: Reduced allowed tolerance, unit_vectors.

rstoneback committed:
ENH: Updated parameters to reduce run-time.

rstoneback committed:
ENH: Restored accuracy of field line trace.

rstoneback committed:
ENH: Restored step_size of unit_vector.

rstoneback committed:
ENH: Added magnetic footpoint location method.

rstoneback committed:
ENH: Removed method no longer needed related to closed field line loop.

rstoneback committed:
ENH: Removed step_until_intersect, related to closed field line loop technique.

rstoneback committed:
ENH: Clarified method name.

rstoneback committed:
ENH: Incorporated footpoint location info.

rstoneback committed:
STY: Updated method names.

rstoneback committed:
STY: Updates

rstoneback committed:
ENH: Incorporated apex_location_info

rstoneback committed:
STY: Updated docstrings.

rstoneback committed:
TES: Removed tests from coverage report.

rstoneback committed:
Update .coveragerc

rstoneback committed:
Merge pull request #17 from pysat/field_trace_fix

BUG/ENH: Improvements to step_until_intersect and bug fix for field_line_trace

rstoneback committed:
Merge branch 'develop' into local_field_vector

rstoneback committed:
ENH: Added d,e vectors as per Richmond notation. Centered difference for 0 apex height change.

rstoneback committed:
Merge pull request #26 from pysat/local_field_vector

ENH: New method for calculating geomagnetic field vectors

rstoneback committed:
ENH: Validated E,D vector calculations.

rstoneback committed:
TES: Added tests for E,D vectors.

rstoneback committed:
ENH/TES: Added direct comparison with apexpy at magnetic equator.

jklenzing committed:
TST: update Travis environment

rstoneback committed:
ENH: Switched output to most accurate calculation for d, e vectors.

rstoneback committed:
TES: Added plot for change in vectors based upon calc method.

rstoneback committed:
ENH: Improved overall robustness. Downselected one path for d,e vectors.

rstoneback committed:
TES/BUG: Fixed apex height plots

rstoneback committed:
Added apexpy for comparison plots

rstoneback committed:
Merge pull request #28 from pysat/jklenzing-patch-1

TST: update Travis environment

rstoneback committed:
ENH: Code reuse in unit vector generation.

rstoneback committed:
STY: Changed keyword to include_debug

rstoneback committed:
STY: Changed keyword to include_debug

rstoneback committed:
STY: Changed keyword to include_debug

rstoneback committed:
BUG: Fixed typo

rstoneback committed:
ENH: Polished unit vector method.

rstoneback committed:
Merge branch 'contravariant' of https://github.com/pysat/pysatMagVect into contravariant

rstoneback committed:
TET: Test cleanup

rstoneback committed:
TET: Added d magnitude normalized difference plots.

rstoneback committed:
ENH: Minor cleanup (check)

rstoneback committed:
BUG/STY: Incorrect plot title. Updated errorbar plot.

rstoneback committed:
Merge branch 'develop' into contravariant

rstoneback committed:
Update .travis.yml

rstoneback committed:
Merge pull request #27 from pysat/contravariant

Contravariant and covariant basis vectors

rstoneback committed:
ENH: Added unit test for change in r when stepping across field lines.

rstoneback committed:
Merge branch 'develop' of https://github.com/pysat/pysatMagVect into scaling_update

rstoneback committed:
BUG: Fixed plot title

rstoneback committed:
TES: Upgrade longitude difference so values are between -180, 180

rstoneback committed:
ENH: Simplified initial zonal, meridional vector generation.

jklenzing committed:
TST: clean up pysat install on travis

rstoneback committed:
TES: Improved organization of test methods.

rstoneback committed:
STY: Improved docstring

rstoneback committed:
STY: Improved docstring

rstoneback committed:
TET: Added D, E stepsize sensitivity tests

rstoneback committed:
ENH: Improved robustness of parameters

rstoneback committed:
ENH: New method for scaling drifts and fields via E and D vectors

rstoneback committed:
STY: Cleaned up docstring

rstoneback committed:
TES: Added unit test for heritage scaling. Corrected test bug.

rstoneback committed:
BUG: Fixes for tests

rstoneback committed:
BUG: Sorting out bugs

rstoneback committed:
TES: Added D, E magnitude plots

rstoneback committed:
ENH: Updated default step size based upon testing

rstoneback committed:
TES: New test covering edge_steps impact on D vector accuracy

rstoneback committed:
TES: tight_layout added to plots in test_core

rstoneback committed:
TES: Updated default parameters.

rstoneback committed:
ENH: Default apex_location_info fine step 1.E-4

rstoneback committed:
ENH: Incorporated latest IGRF model

rstoneback committed:
STY: Changed filename

rstoneback committed:
ENH: Updated calls to IGRF13

rstoneback committed:
ENH: Update setup for new IGRF

rstoneback committed:
BUG: Updated for IGRF13

rstoneback committed:
ENH: Removed pysat dependency

rstoneback committed:
ENH: generalized methods to **kwargs

rstoneback committed:
TES: First basic tests for satellite code

rstoneback committed:
TES: Expanded satellite testing

rstoneback committed:
BUG: Updated tests

rstoneback committed:
TES: One more test

rstoneback committed:
ENH: Updated changelog

rstoneback committed:
TES: Added verification meridional direction along maximum apex height gradient

rstoneback committed:
STY: Cleaned up imports

rstoneback committed:
changelog update

rstoneback committed:
changelog update

rstoneback committed:
changelog update

rstoneback committed:
BUG: Covered plots for travis and python2

rstoneback committed:
ENH: Implemented potentially higher accuracy method for d vector meridional magnitude

rstoneback committed:
ENH: Removed new path as it did not improve accuracy.

rstoneback committed:
Merge pull request #35 from pysat/scaling_update

Closing Remaining Tasks on Multipole Generalization

rstoneback committed:
Merge branch 'develop' into travis_update

rstoneback committed:
Update .travis.yml

rstoneback committed:
Update .travis.yml

rstoneback committed:
Update .travis.yml

rstoneback committed:
Update .travis.yml

rstoneback committed:
Update .travis.yml

rstoneback committed:
Update .travis.yml

rstoneback committed:
Update .travis.yml

rstoneback committed:
Switched to next() from .next()

rstoneback committed:
Switched to next() from .next()

rstoneback committed:
Merge pull request #34 from pysat/travis_update

TST: clean up pysat install on travis

rstoneback committed:
Replaced .ix calls with .loc

rstoneback committed:
Replaced .netx() calls with next() calls

rstoneback committed:
Update version.

rstoneback committed:
Reduced default testing range.

rstoneback committed:
Merge pull request #36 from pysat/test_corrections

Python 3 Test Compatibility

rstoneback commented:
I'm planning on releasing this version next week. Pinging @aburrell and @jklenzing as an FYI. I'm certainly open to code review though given the size of the pull it is not expected.

aburrell commented on pysatMagVect/tests/test_comparison.py:
```diff
@@ -0,0 +1,212 @@
+        return
+
+
+    def test_apexpy_v_pymv(self):
```
Missing docstrings

rstoneback replied:
Added

aburrell commented on pysatMagVect/tests/test_comparison.py:
```diff
@@ -0,0 +1,212 @@
+        dotmagvect_mer = me*fe + mn*fn + mu*fu
+
+
+        try:
```
Is making a figure the most efficient way to do a unit test?

rstoneback replied:
For this package unit test plots are the way to go. I've used every plot to get the code to where it is at.

rstoneback replied:
Added issue #38 to ensure this is resolved.

aburrell commented on pysatMagVect/_core.py:
```diff
@@ -109,10 +109,12 @@ def geodetic_to_ecef(latitude, longitude, altitude):
+ecef_to_geodetic = pysatMagVect.fortran_coords.ecef_to_geodetic
 
-def ecef_to_geodetic(x, y, z, method=None):
+def python_ecef_to_geodetic(x, y, z, method=None):
```
Curious about this name change

rstoneback replied:
I added a ecef_to_geodetic function written in Fortran that I wanted the code to use by default. I kept the python version to check accuracy and it is easier to read.

aburrell commented on pysatMagVect/_core.py:
```diff
@@ -960,331 +1389,345 @@ def closed_loop_edge_lengths_via_footpoint(glats, glons, alts, dates, direction,
     as well as at the magnetic equator.
-    
+
     All inputs are assumed to be 1D arrays.
```
should be shorter

aburrell reviewed (commented):
Some general notes and one question.

jklenzing commented on .travis.yml:
```diff
@@ -1,44 +1,44 @@
+  include:
+    - python: 2.7
+    - python: 3.6
+    - python: 3.7
```
Should we add python 3.8?

rstoneback replied:
Added

jklenzing commented on pysatMagVect/_core.py:
```diff
@@ -23,7 +23,7 @@
 
```
2 blank lines before functions

rstoneback replied:
Fixed.

jklenzing commented on pysatMagVect/_core.py:
```diff
@@ -90,12 +90,12 @@ def geodetic_to_ecef(latitude, longitude, altitude):
+
     """
 
 
```
should be one blank line

jklenzing commented on pysatMagVect/_core.py:
```diff
@@ -177,11 +179,11 @@ def ecef_to_geodetic(x, y, z, method=None):
 
 def enu_to_ecef_vector(east, north, up, glat, glong):
```
2 blank lines before functions, here and throughout

rstoneback replied:
Fixed

jklenzing commented on pysatMagVect/_core.py:
```diff
@@ -346,87 +348,103 @@ def field_line_trace(init, date, direction, height, steps=None,
+    # ensure date is a float for IGRF call
     if not isinstance(date, float):
         # recast from datetime to float, as required by IGRF12 code
         doy = (date - datetime.datetime(date.year,1,1)).days
```
missing whitespace after ','

rstoneback replied:
fixed

jklenzing commented on pysatMagVect/_core.py:
```diff
@@ -541,31 +569,27 @@ def calculate_mag_drift_unit_vectors_ecef(latitude, longitude, altitude, datetim
     be = [];
```
semicolons?

rstoneback replied:
removed

jklenzing commented on pysatMagVect/_core.py:
```diff
@@ -541,31 +569,27 @@ def calculate_mag_drift_unit_vectors_ecef(latitude, longitude, altitude, datetim
-                                        step_size=step_size, max_steps=max_steps)
+        trace = full_field_line(init, time, ref_height, step_size=step_size,
+                                                        max_steps=max_steps,
+                                                        steps=steps)
```
over-indentation

jklenzing commented on pysatMagVect/_core.py:
```diff
@@ -600,353 +624,758 @@ def calculate_mag_drift_unit_vectors_ecef(latitude, longitude, altitude, datetim
+                'e_mer_z' : e_mer_z,
+                'e_fa_x' : e_fa_x,
+                'e_fa_y' : e_fa_y,
+                'e_fa_z' : e_fa_z,
```
whitespace before ':'

jklenzing commented on pysatMagVect/_core.py:
```diff
@@ -600,353 +624,758 @@ def calculate_mag_drift_unit_vectors_ecef(latitude, longitude, altitude, datetim
+                    'd_zon2_z' : d_zon2_z,
+                    'd_mer2_x' : d_mer2_x,
+                    'd_mer2_y' : d_mer2_y,
+                    'd_mer2_z' : d_mer2_z,
```
whitespace before ':'

jklenzing commented on pysatMagVect/_core.py:
```diff
@@ -600,353 +624,758 @@ def calculate_mag_drift_unit_vectors_ecef(latitude, longitude, altitude, datetim
+
+
+
+def step_along_mag_unit_vector(x, y, z, date, direction=None, num_steps=1.,
```
Too many blank lines

rstoneback replied:
Fixed

jklenzing commented on pysatMagVect/_core.py:
```diff
@@ -600,353 +624,758 @@ def calculate_mag_drift_unit_vectors_ecef(latitude, longitude, altitude, datetim
-def apex_location_info(glats, glons, alts, dates):
+        x = x + step_size*ux#*scalar
+        y = y + step_size*uy#*scalar
+        z = z + step_size*uz#*scalar
```
Are these commented bits from an old version?  Or inline comments?

jklenzing commented on pysatMagVect/_core.py:
```diff
@@ -600,353 +624,758 @@ def calculate_mag_drift_unit_vectors_ecef(latitude, longitude, altitude, datetim
+
+    """
+
+   # use input location and convert to ECEF
```
indentation

jklenzing commented on pysatMagVect/_core.py:
```diff
@@ -600,353 +624,758 @@ def calculate_mag_drift_unit_vectors_ecef(latitude, longitude, altitude, datetim
+        trace_north = field_line_trace(root, double_date, 1., 120.,
+                                    steps=steps,
+                                    step_size=step_size,
+                                    max_steps=num_steps)
```
indentation doesn't match paren (same in southern function below

jklenzing commented on pysatMagVect/_core.py:
```diff
@@ -600,353 +624,758 @@ def calculate_mag_drift_unit_vectors_ecef(latitude, longitude, altitude, datetim
+                                    vector_direction, step_size=None,
+                                    max_steps=None, steps=None,
+                                    edge_length=25., edge_steps=5,
+                                    ecef_input=False):
```
indentation doesn't match paren

jklenzing commented on pysatMagVect/_core.py:
```diff
@@ -960,331 +1389,345 @@ def closed_loop_edge_lengths_via_footpoint(glats, glons, alts, dates, direction,
+
+    # D and E vectors at user supplied location
+    # good for mapping to magnetic equator
+    _, _, _, _, _, _, _, _, _, infod = calculate_mag_drift_unit_vectors_ecef(ecef_xs, ecef_ys, ecef_zs, dates,
```
This function is defined above and is called like this every time.  Could we reconfigure the returned variables to aid in clarity?

rstoneback replied:
It is used elsewhere (step_along_..) with the unit vectors outputs assigned to variable names. More importantly, there is downstream code that depends upon pysatMagVect and changing the outputs here causes a ripple of work. Instead, I added a new method (calculate_geomagnetic_basis) with limited input options and an output dictionary that just transparently calls the calculate_mag_drift_unit_vectors. Existing code gets to remain the same and new code can use the simpler, clearer interface.

jklenzing commented on pysatMagVect/satellite.py:
```diff
@@ -264,7 +261,7 @@ def add_mag_drift_unit_vectors(inst, max_steps=40000, step_size=10.):
                                           'of the IVM also expressed in ECEF (sc_*hat_*).'),
```
indentation

jklenzing commented on pysatMagVect/satellite.py:
```diff
@@ -345,26 +342,26 @@ def add_mag_drift_unit_vectors(inst, max_steps=40000, step_size=10.):
+
     inst['iv_zon'] = {'data':inst['unit_zon_x'] * inst['iv_x'] + inst['unit_zon_y']*inst['iv_y'] + inst['unit_zon_z']*inst['iv_z'],
                       'units':'m/s',
                       'long_name':'Zonal ion velocity',
```
whitespace after ':'

jklenzing commented on pysatMagVect/tests/test_apex.py:
```diff
@@ -0,0 +1,716 @@
+        try:
+
+
+            plt.figure()
```
too many blank lines

jklenzing commented on pysatMagVect/tests/test_transformations_and_tracing.py:
```diff
@@ -0,0 +1,789 @@
+# results from omniweb calculator
+omni_list = [[550. , 20.00   , 0.00 , 29.77,  359.31,  -9.04 ,   3.09],
+    [550. , 20.00   , 7.50 , 29.50  ,  7.19 , -8.06 ,   9.54],
+    [550. , 20.00   ,15.00 , 28.34  , 15.01 , -7.51 ,  16.20],
```
indentation

jklenzing commented on pysatMagVect/tests/test_transformations_and_tracing.py:
```diff
@@ -0,0 +1,789 @@
+#                 apex_z[i,:-1] = np.abs(z2 - z)
+#
+#
+#         else:
```
Do we need commented code?

jklenzing commented on setup.py:
```diff
@@ -27,26 +23,23 @@
-numpy.distutils.core.setup( 
-# setuptools.setup( 
+numpy.distutils.core.setup(
+# setuptools.setup(
```
commented code

jklenzing commented on setup.py:
```diff
@@ -1,22 +1,18 @@
+# Copyrighted by The University of Texas at Dallas subject to the United States
 # government unlimited rights (DFAR 252.227-7013-5).
 
 import setuptools
```
package not used

rstoneback replied:
Turns put python setup.py develop needs this import statement to work, even though the package isn't obviously used within setup.py

jklenzing commented on setup.py:
```diff
@@ -1,22 +1,18 @@
-    sources = [os.path.join('pysatMagVect', 'igrf12.f')]
-    )
+extensions = [Extension(name='pysatMagVect.igrf',
+                        sources = [os.path.join('pysatMagVect', 'igrf13.f')]),
```
no whitespace around '=' for kwargs

jklenzing commented on setup.py:
```diff
@@ -1,22 +1,18 @@
+extensions = [Extension(name='pysatMagVect.igrf',
+                        sources = [os.path.join('pysatMagVect', 'igrf13.f')]),
+              Extension(name='pysatMagVect.fortran_coords',
+                        sources = [os.path.join('pysatMagVect', '_coords.f')])]
```
same

jklenzing commented on setup.py:
```diff
@@ -27,26 +23,23 @@
+# setuptools.setup(
     name = 'pysatMagVect',
-    version = version,        
+    version = version,
```
no whitespace

jklenzing reviewed (commented):
Mostly style comments caught by flake8.  I'd recommend moving plots out of unit tests and into a demo folder to speed up travis.

rstoneback committed:
TES: Docstring and formatting improvements

rstoneback committed:
STY: Updated code formatting

rstoneback committed:
Merge branch 'test_corrections' into develop

rstoneback committed:
ENH: Added python 3.8 to unit tests

rstoneback committed:
STY: Shortened docstring

rstoneback committed:
STY: 2 lines between methods

rstoneback committed:
STY: Improved formatting

rstoneback committed:
STY: Removed semicolons

rstoneback committed:
STY: Updated formatting

rstoneback committed:
STY: Updated formatting

rstoneback committed:
STY: Updated formatting

rstoneback committed:
STY: Restored setup.py

rstoneback committed:
ENH: Added simple interface for obtaining basis vectors

rstoneback committed:
TES: Added coverage for simple interface run

rstoneback commented:
That should cover most everything:
 - Answered the questions
 - Added docstrings to test code
 - Made numerous formatting improvements
 - Moved Notes to after Return in docstrings
 - Added new method with a single output dict for the geomagnetic basis vectors. Existing method kept for compatibility of downstream code.
 - One request not currently filled. The unit tests are still making plots. The figures are a big part of testing for this package. It is also true that when I run testing locally I set it to high resolution and use the plots in the developing manuscript.

rstoneback committed:
STY: Moved Notes to after Return

rstoneback commented:
Thanks for the review @aburrell and @jklenzing . Let's talk about the plots tomorrow. Cheers.

rstoneback committed:
STY: Updated description

rstoneback committed:
STY: Removed tests from pip distributions via manifest.in

rstoneback commented:
I added issue #38 to improve the unit tests by removing the plots. I also modified manifest.in to exclude the test directory for pip produced distributions. Standard users won't see the unit tests. I plan on moving forward with the release. Thanks again for the feedback.
